### ISSUE TITLE
Error When Adding a Note in SF Lightning Due to DLRS Trigger

### ISSUE THREAD

**Issue by ccharles76** [afawcett/declarative-lookup-rollup-summaries#558]
I need to track employee/consultant expenses and have a custom object in Salesforce Enterprise Edition called “Expenses”.  We need to be able to see if there are attachments (receipts) to those expenses.  We are using the Lightning interface, so we had to use the new Salesforce Files/Notes object instead of the traditional Attachments related list.  Because there is not a master-detail relationship between expenses and Files, we were unable to do a roll-up summary and thus I used the Declarative Rollup Lookup Summary tool to create a rollup summary (see attached screen shot of the setup).

It is calculating the number of “Files” correctly and I thought all was well.  However, I recently learned that the trigger that the DRLS created causes an issue with the “Files” / “Notes” object.  In Lightning or Classic, if I attempt to create a new Note in Salesforce, I receive an error.  The error is more descriptive in Classic and the error says: 

Apex trigger dlrs_ContentDocumentLinkTrigger caused an unexpected exception, contact your administrator: dlrs_ContentDocumentLinkTrigger: execution of AfterInsert

caused by: System.QueryException: Implementation restriction: ContentDocumentLink requires a filter by a single Id on ContentDocumentId or LinkedEntityId using the equals operator or multiple Id's using the IN operator.: Class.dlrs.LREngine.QueryExecutor.query: line 659, column 1

Does anyone have a suggestion as to how to solve this issue?  Thank you, in advance!

![dlrs setup](https://user-images.githubusercontent.com/31748043/30188081-d0e598b8-93e9-11e7-9aea-383b1588e90f.PNG)
![dlrs trigger](https://user-images.githubusercontent.com/31748043/30188083-d34c2964-93e9-11e7-9023-13433b09fd7b.PNG)

**Comment by ccharles76** [comment]
Sorry that I'm new to this, but has anyone experienced this issue before or know how to overcome it?  Thank you!

**Comment by aheber** [comment]
@ccharles76 I haven't used DLRS for this object before. The error is strange indeed. I'll make some time this week to try this out. If I can get the same error then we can work on a resolution.

I'll respond back at some point this week.

**Comment by ccharles76** [comment]
I was able to find a work-around, but it doesn't work as well as I would like.  I wanted to share it:
There is a small issue with my work-around that I could use some advise on. I removed the trigger and revised the DLRS record by adding a Lookup Rollup Summary Unique Name and changing the Calculation Mode to Process Builder (as shown in the file attachment; Revised DLRS Setup). 

![revised dlrs setup](https://user-images.githubusercontent.com/31748043/30618027-f41b86ec-9d55-11e7-8025-ef2dbd5d27ae.PNG)

I was able to create a Process Builder rule based on my Expense Object when the Amount field Is Changed = TRUE OR Amount Is Null = FALSE. The Immediate Action is to call APEX, referencing my rollup summary unique name and the Parent Record Id of my Expense.Id field.

![pbuilder-action-callapex](https://user-images.githubusercontent.com/31748043/30618040-fecfe574-9d55-11e7-9e9e-55647528ab54.PNG)
![pbuilder-amountchanged](https://user-images.githubusercontent.com/31748043/30618039-fecd31a8-9d55-11e7-8f07-8bb6ce375858.PNG)

It runs correctly after I make an edit to the Expense record, but if I just add a File (attachment), the process builder does not trigger.
Any suggested way to get the Process to fire upon uploading of a File to the expense record?
